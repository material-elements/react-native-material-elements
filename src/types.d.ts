export * from './components/types';
export * from './types/index';
