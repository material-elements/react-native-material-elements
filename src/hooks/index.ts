export * from './useRestyle';
export * from './useThemedProps';
