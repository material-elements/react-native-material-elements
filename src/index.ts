export * from './components';
export * from './hooks';
export * from './libraries';
export * from './utils';
