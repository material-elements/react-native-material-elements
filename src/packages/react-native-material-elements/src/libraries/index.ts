export * from './themes';
export * from './style';
