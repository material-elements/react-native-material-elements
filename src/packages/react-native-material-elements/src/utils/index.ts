export * from './styleGenerator';
export * from './utils';
export * from './dimensions';
