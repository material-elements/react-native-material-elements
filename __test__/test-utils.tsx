import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';

export const ThemeWrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: ThemeWrapper, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
