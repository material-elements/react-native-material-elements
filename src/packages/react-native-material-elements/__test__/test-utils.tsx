import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';
import { ThemeProviderProps } from '../src/libraries/types';

interface CustomRenderOptions extends Omit<RenderOptions, 'wrapper'> {
  themeProps?: Partial<ThemeProviderProps>;
}

export const ThemeWrapper: React.FC<ThemeProviderProps> = ({ children, ...props }) => {
  return <ThemeProvider {...props}>{children}</ThemeProvider>;
};

export const customRender = (ui: React.ReactElement, options?: CustomRenderOptions) => {
  const { themeProps, ...rest } = options || {};
  return render(ui, {
    wrapper: wrapperProps => <ThemeWrapper {...themeProps}>{wrapperProps.children}</ThemeWrapper>,
    ...rest,
  });
};

export * from '@testing-library/react-native';
export { customRender as render };
