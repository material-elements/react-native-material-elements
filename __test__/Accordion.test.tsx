import React from 'react';
import { render, waitFor } from './test-utils';
import { Accordion, AccordionSummary, Text } from '../src';
import { View, StyleSheet } from 'react-native';

describe('Accordion Component', () => {
  const mockAccordionTestId = 'accordion-test-id';

  const mockRef = React.createRef<View>();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', async () => {
    const { toJSON } = render(<Accordion />);
    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('should forward ref correctly', () => {
    render(<Accordion ref={mockRef} />);
    expect(mockRef.current).toBeDefined();
    expect(mockRef.current).toBeInstanceOf(View);
  });

  it('should change the border radius when passed the square prop', () => {
    const { getByTestId } = render(<Accordion testID={mockAccordionTestId} square />);

    const accordion = getByTestId(mockAccordionTestId);
    expect(accordion.props.style).not.toHaveProperty('borderRadius');
  });

  it('should change the opacity when passed the disable prop', () => {
    const { getByTestId } = render(<Accordion testID={mockAccordionTestId} disable />);

    const accordion = getByTestId(mockAccordionTestId);
    const flattenedStyle = StyleSheet.flatten(accordion.props.style);

    expect(flattenedStyle.opacity).toEqual(0.5);
  });
});

describe('AccordionSummary Component', () => {
  const mockRef = React.createRef<View>();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', async () => {
    const { toJSON } = render(<AccordionSummary />);
    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('should forward ref correctly', () => {
    render(<AccordionSummary ref={mockRef} />);
    expect(mockRef.current).toBeDefined();
    expect(mockRef.current).toBeInstanceOf(View);
  });

  it('should render the icon when passed the expandIcon prop', () => {
    const { getByText } = render(<AccordionSummary expandIcon={<Text>expandIcon</Text>} />);

    const expandIcon = getByText('expandIcon');
    expect(expandIcon).toBeDefined();
  });
});
