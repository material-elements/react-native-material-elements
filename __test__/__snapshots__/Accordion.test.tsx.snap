// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component should render correctly 1`] = `
<View
  style={
    [
      {},
      {},
      [
        {
          "backgroundColor": "#EEEEEE",
          "borderRadius": 8,
          "opacity": 1,
          "width": "100%",
        },
        {},
        undefined,
      ],
    ]
  }
/>
`;

exports[`AccordionSummary Component should render correctly 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 13,
          "width": "100%",
        },
        {},
        {},
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "alignContent": "center",
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "width": "90%",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          [
            {},
            {},
            [
              {
                "flex": 1,
              },
              undefined,
            ],
          ]
        }
      />
    </View>
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "transform": [
            {
              "rotate": "0deg",
            },
          ],
          "width": "10%",
        }
      }
    />
  </View>
  <View
    collapsable={false}
    style={
      {
        "height": 0,
        "opacity": 0,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        [
          {
            "width": "100%",
          },
          {
            "height": null,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      {
        "opacity": 0,
        "position": "absolute",
        "zIndex": -1,
      }
    }
  />
</View>
`;
